lib/archethic/db/embedded_impl/chain_reader.ex

```diff
@@ -128,33 +128,74 @@ defmodule Archethic.DB.EmbeddedImpl.ChainReader do
     start = System.monotonic_time()
 
-    case ChainIndex.get_tx_entry(address, db_path) do
-      {:error, :not_exists} ->
-        {[], false, ""}
+    genesis_address = ChainIndex.get_genesis_address(address, db_path)
+    filepath = ChainWriter.chain_path(db_path, genesis_address)
 
-      {:ok, %{genesis_address: genesis_address}} ->
-        filepath = ChainWriter.chain_path(db_path, genesis_address)
-        fd = File.open!(filepath, [:binary, :read])
+    if File.exists?(filepath) do
+      fd = File.open!(filepath, [:binary, :read])
 
-        {transactions, more?, paging_state} =
-          case Keyword.get(opts, :order, :asc) do
-            :asc ->
-              process_get_chain(fd, address, fields, opts, db_path)
+      {transactions, more?, paging_state} =
+        case Keyword.get(opts, :order, :asc) do
+          :asc ->
+            process_get_chain(fd, fields, opts, db_path)
+
+          :desc ->
+            process_get_chain_desc(fd, genesis_address, fields, opts, db_path)
+        end
 
-            :desc ->
-              process_get_chain_desc(fd, genesis_address, fields, opts, db_path)
+      # we want different metrics for ASC and DESC
+      :telemetry.execute([:archethic, :db], %{duration: System.monotonic_time() - start}, %{
+        query:
+          case Keyword.get(opts, :order, :asc) do
+            :asc -> "get_transaction_chain"
+            :desc -> "get_transaction_chain_reverse"
           end
+      })
 
-        # we want different metrics for ASC and DESC
-        :telemetry.execute([:archethic, :db], %{duration: System.monotonic_time() - start}, %{
-          query:
-            case Keyword.get(opts, :order, :asc) do
-              :asc -> "get_transaction_chain"
-              :desc -> "get_transaction_chain_reverse"
-            end
-        })
+      {transactions, more?, paging_state}
+    else
+      {[], false, nil}
+    end
+  end
+
+  @doc """
+  Stream chain tx from the beginning until a given limit address
+  """
+  @spec stream_chain(
+          genesis_address :: binary(),
+          fields :: list(),
+          db_path :: binary()
+        ) :: Enumerable.t()
+  def stream_chain(genesis_address, fields, db_path) do
+    filepath = ChainWriter.chain_path(db_path, genesis_address)
+
+    case File.open(filepath, [:binary, :read]) do
+      {:ok, fd} ->
+        Stream.resource(
+          fn -> process_get_chain(fd, fields, [], db_path) end,
+          fn
+            {transactions, true, paging_state} ->
+              next_transactions =
+                process_get_chain(
+                  fd,
+                  fields,
+                  [paging_state: paging_state],
+                  db_path
+                )
+
+              {transactions, next_transactions}
+
+            {transactions, false, _} ->
+              {transactions, :eof}
+
+            :eof ->
+              {:halt, nil}
+          end,
+          fn _ -> :file.close(fd) end
```

Comment:
you used File.open from elixir and here :file from erlang, maybe it's better to use the same (probably FIle is just a wrapper around :file but it's better to use the same for consistency).